### PR TITLE
cork/uncork logic client fix

### PR DESCRIPTION
**Pull request by scarry1992**

### CODE REVIEW

File: stream.js [hunk 1]
```diff
@@ -73,8 +73,14 @@ function WebSocketStream(target, protocols, options) {
   if (socket.readyState === socket.OPEN) {
     stream = proxy
   } else {
-    stream = duplexify.obj()
-    socket.onopen = onopen
+    if (isBrowser) {
```
Comment (mcollina) [inline stream.js]: you can collapse this into the previous else.

File: stream.js [hunk 2]
```diff
@@ -73,8 +73,14 @@ function WebSocketStream(target, protocols, options) {
...
+      stream = proxy
+      stream.cork()
+      socket.onopen = onopenBrowser
```
Comment (mcollina) [inline stream.js]: can the same logic be applied to Node as well? What will cause issues?

Comment (scarry1992) [reply stream.js]: I'll try to use same logic for Node and browser, but for Node needed duplexify logic for destroy() method as I see.

File: stream.js (hunk 1 above)
Comment (scarry1992) [reply stream.js]: You mean something like this:

  if (isNative && isBrowser) {
      socket = new WS(target, protocols)
      stream = proxy
      stream.cork()
      socket.onopen = onopenBrowser
    } else {
      socket = new WS(target, protocols, options)
      stream = duplexify.obj()
      socket.onopen = onopen
    }

Comment (mcollina) [reply stream.js]:
```js
if (socket.readyState === socket.OPEN) {
    stream = proxy
  } else if (isBrowser) {

} else {

}
```

**Comment by scarry1992** [comment]
@mcollina, if structure fixed

**Comment by mcollina** [comment]
I don't really understand why we can't use the same logic for Node as well.

**Comment by scarry1992** [comment]
@mcollina, so what about that PR?

**Comment by scarry1992** [comment]
@mcollina, do you release it in npm?

**Comment by mcollina** [comment]
Yes, v5.2.0.

**Pull request merged**